### PR TITLE
feat(app): add ab framework to app

### PR DESCRIPTION
**Pull request by danilowoz**
This introduces the implementation of the AB framework across the apps (homepage and editor) in order to use the same user-id in both environments and it also persists the id from an anonymous user to a logged-in status. 

**Implementation todo**
- [x] Editor
- [x] First experiment candidate https://github.com/codesandbox/codesandbox-client/pull/5855
- [x] Homepage

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5898/commits/6ffc8d2f474340709dadefd3324d0d9abedf2623 is at https://pr5898.build.csb.dev/s/new.

File: packages/app/src/app/pages/Sandbox/Editor/Header/SignInAd.tsx [hunk 1]
```diff
@@ -4,12 +4,25 @@ import { Stack, Text, Button } from '@codesandbox/components';
...
   const { signInClicked } = useActions();
   const [show, setShow] = useState(false);
 
+  const experimentPromise = useExperimentResult('editor-signin-banner-trigger');
+
   useEffect(() => {
+    experimentPromise.then(experiment => {
+      if (experiment === ExperimentValues.A) {
+        // eslint-disable-next-line no-console
+        console.log('A');
```
Comment (CompuIves) [inline packages/app/src/app/pages/Sandbox/Editor/Header/SignInAd.tsx:19]: Some remaining console.logs 😄

File: packages/common/src/config/env.ts [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
 
+export const AB_TESTING_URL = 'https://ab-testing.codesandbox.workers.dev';
+
+export const getExperimentUserId = () => {
+  const KEY_NAME = 'csb-ab-user-id';
+
+  let userId = localStorage.getItem(KEY_NAME);
+
+  if (!userId) {
+    userId = uuid.v4();
```
Comment (CompuIves) [inline packages/common/src/config/env.ts:19]: Great!

**Review by CompuIves — APPROVED** [review]
Nice!!! Looks perfect 😄

**Pull request merged**